src/estimation/src/BerdyHelper.cpp

```diff
@@ -1526,6 +1590,19 @@ bool BerdyHelper::updateKinematicsFromFloatingBase(const JointPosDoubleArray& jo
     m_jointPos = jointPos;
     m_jointVel = jointVel;
 
+    // Update kinDyn computation
+    iDynTree::KinDynComputations kinDynComputations;
+    kinDynComputations.loadRobotModel(m_model);
+    kinDynComputations.setRobotState(m_jointPos,
+                                     m_jointVel,
+                                     m_gravity);
```

Comment:
Allocating a complete `iDynTree::KinDynComputation` will slow down and make this code extremly non-real-time safe. For computing the forward kinematics of the link, it should be sufficient to just call the [`ForwardPositionKinematics`](https://github.com/robotology/idyntree/blob/master/src/model/include/iDynTree/Model/ForwardKinematics.h#L44) function.